api/views.py

```diff
@@ -0,0 +1,161 @@
+    def fire(self, request, *args, **kwargs):
+        employee = self.get_object()
+        employee.hired = False
+        employee.save(update_fields=['hired'])
+        serializer = EmployeeSerializer(employee, many=False)
+        serializer.is_valid(raise_exception=True)
+        self.perform_update(serializer)
+        return Response(serializer.data)
+
+    @action(detail=True, methods=['get'])
+    def hire_again(self, request, *args, **kwargs):
+        employee = self.get_object()
+        employee.hired = True
+        employee.save(update_fields=['hired'])
+        serializer = EmployeeSerializer(employee, many=False)
+        serializer.is_valid(raise_exception=True)
+        self.perform_update(serializer)
+        return Response(serializer.data)
+
+    @action(detail=True, methods=['post'])
+    def change_position(self, request, *args, **kwargs):
+        pesel = request.POST.get('pesel')
+        new_position = request.POST.get('new_position')
+
+        if pesel is not None and new_position is not None:
+            employee = get_object_or_404(Employee, pesel=pesel)
+            employee.position = new_position
+            employee.save(update_fields=['position'])
+            serializer = EmployeeSerializer(employee, many=False)
+            serializer.is_valid(raise_exception=True)
+            self.perform_update(serializer)
+            return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def summary_cost_of_salaries(self, request, *args, **kwargs):
+        sum_of_salaries = Employee.objects.aggregate(sum_of_salaries=Sum('salary'))
+        return Response("Summary cost of all salaries: " + str(sum_of_salaries['sum_of_salaries']))
+
+
+class IncomeViewSet(viewsets.ModelViewSet):
+    queryset = Income.objects.all()
+    serializer_class = IncomeSerializer
+    filter_fields = ('name', 'date')
+
+    @action(detail=False, methods=['get'])
+    def last_incomes(self, request, *args, **kwargs):
+        number_of_incomes_to_show = request.GET.get('number_of_incomes')
+        incomes = Income.objects.order_by('date')
+        if number_of_incomes_to_show:
+            incomes = incomes[:int(number_of_incomes_to_show)]
+
+        serializer = EventSerializer(incomes, many=True)
+        return Response(serializer.data)
+
+    @action(detail=False, methods=['get'])
+    def sum(self, request, *args, **kwargs):
+        days_back = request.GET.get('days', 7)
+        if str(days_back).isdigit():
+            current_date = datetime.today().date()
+            date_for_filter = current_date - timedelta(days=int(days_back))
+            sum_of_incomes = Income.objects.filter(date__gte=date_for_filter).aggregate(sum_of_incomes=Sum('sum'))[
+                'sum_of_incomes']
+            if not sum_of_incomes:
+                sum_of_incomes = "0"
+            return Response("Total income from the last " + str(days_back) + " days is: " + str(sum_of_incomes))
```

Comment:
`sum_of_incomes` can not be falsy here in this context. If you want to be sure, then just do:

```python
Income
	.objects
	.filter(date__gte=date_for_filter)
	.aggregate(sum_of_incomes=Sum('sum'))
	.get('sum_of_incomes', 0)
```

and then this whole `if not sum_of_incomes` if statement is unnecessary.

Also, this is more pythonic way:

```python
"Total income from the last {} days is: {}".format(days_back, sum_of_incomes)
```

or even better, use `f""` strings (they were introduced in Python 3.6):

```python
f"Total income from the last {days_back} days is: {sum_of_incomes}"
```

Reply:
Fix this in all places, please.

Reply:
done